src/internal/shoelace-element.ts

```diff
@@ -92,6 +92,58 @@ export default class ShoelaceElement extends LitElement {
 
     return event as GetCustomEventType<T>;
   }
+
+  // @ts-expect-error
+  static version: string = typeof shoelaceVersion !== "undefined" ? shoelaceVersion : ""
+
+  static define (name: string) {
+    define(name, this)
+  }
+
+  static scopedElements: Record<string, typeof ShoelaceElement> = {}
+
+  connectedCallback () {
+    super.connectedCallback()
+    Object.entries((this.constructor as typeof ShoelaceElement).scopedElements).forEach(([name, component]) => {
+      define(name, component)
+    })
+  }
+}
+
+function define (name: string, elementConstructor: CustomElementConstructor | typeof ShoelaceElement) {
+  const currentElementConstructor = window.customElements.get(name) as CustomElementConstructor | typeof ShoelaceElement
+
+  if (!currentElementConstructor) {
+    // @ts-expect-error
+    window.customElements.define(name, toAnonymousClass(elementConstructor))
+    return
+  }
+
+  let newVersion = " "
+  let existingVersion = " "
+
+  if ("version" in elementConstructor) {
+    newVersion += "v" + elementConstructor.version
```

Comment:
We don't really use the `v2.x.x` syntax (except for git tags). Might not matter, but worth calling it out.